src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -627,6 +628,15 @@ final case class AgreementApiServiceImpl(
           stamps = stamps.toManagement
         )
 
+    def validateConsumerEmails(agreement: PersistentAgreement): Future[Unit] =
+      for {
+        consumer <- tenantManagementService
+          .getTenantById(agreement.consumerId)
+        _        <- Future
+          .failed(ConsumerWithNotValidEmail(agreement.id, agreement.consumerId))
+          .whenA(consumer.mails.filter(_.kind == PersistentTenantMailKind.ContactEmail).isEmpty)
```

Comment:
don't use cats with future
```suggestion
        _        <- if (consumer.mails.filter(_.kind == PersistentTenantMailKind.ContactEmail).isEmpty) 
        Future.failed(ConsumerWithNotValidEmail(agreement.id, agreement.consumerId))
        else Future.unit
```